portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/oauth2/OAuth2.testcase

```diff
@@ -22,242 +22,127 @@ definition {
-
-			ServiceAccessPolicy.editServiceAccessPolicy(
-				enabled = "true",
-				policyName = "OAUTH2_read.portalversion");
-		}
+	@description = "This is a use case for LPS-126597. Authorize external application with remember device enabled."
+	@priority = "5"
+	test ApplicationCanBeAuthorizedWithRememberDeviceEnabled {
+		property test.name.skip.portal.instance = "OAuth2#ExternalApplicationCanBeAuthorizedAndRememberDeviceShouldBeEnabled";
 
-		task ("Assert portal version equals 7.4") {
-			SystemSettings.assertPortalVersion(
-				accessToken = "${accessToken}",
-				portalVersion = "7.4");
+		task ("Add and edit external application with remember device enabled") {
+			OAuth2.addAndEditExternalApplication(
+				applicationName = "Third-party Application",
+				callbackURI = '''http://martamedio.com/oauth2-tester/?url=${portalURL}&client_id=abc123''',
+				checkboxCheckList = "Remember Device",
+				clientId = "abc123",
+				clientProfile = "Web Application",
+				clientSecret = "abc123");
 		}
 
-		task ("Edit SAP") {
-			ServiceAccessPolicy.editServiceAccessPolicy(
-				methodName = "getBuildNumber",
-				newName = "OAUTH2_read.buildnumber",
-				policyName = "OAUTH2_read.portalversion",
-				title = "read Portal build number");
+		task ("Authorize my application with remember device == true") {
+			OAuth2.authorizeMyApplication(
+				clientId = "abc123",
+				clientSecret = "abc123",
+				rememberDevice = "true");
 		}
 
-		task ("View and edit OAuth2 scopes") {
-			Navigator.openSpecificURL(url = "${baseURL}/group/control_panel/manage?p_p_id=com_liferay_oauth2_provider_web_internal_portlet_OAuth2AdminPortlet");
-
-			OAuth2.viewScopes(
-				applicationName = "OAuth Application",
-				removedScope = "read.portalversion",
-				resourcePanels = "Portal Services");
+		// Re-open test harness to check if Device is remembered
 
-			OAuth2.editScopes(
-				applicationName = "OAuth Application",
-				resourceCheckList = "liferay-json-web-services.read.buildnumber",
-				resourcePanels = "Portal Services");
+		task ("Authorize my application") {
+			OAuth2.authorizeMyApplication(
+				clientId = "abc123",
+				clientSecret = "abc123",
+				noAuthorize = "true");
 		}
 
-		task ("Assert portal version equals 7.4") {
-			var accessToken = OAuth2.getAccessToken(
-				clientId = "${generatedClientId}",
-				clientSecret = "${generatedClientSecret}",
-				grantFlow = "clientCredentials");
+		// Execute a request as final verification
 
-			SystemSettings.assertPortalBuildNumber(
-				accessToken = "${accessToken}",
-				portalBuild = "74");
+		task ("Launch request") {
+			OAuth2.launchRequest();
 		}
 	}
 
-	@description = "This is a use case for OAUTH2-242. Assert authorization code works in cluster."
+	@description = "This is a use case for OAUTH2-242. Assert authorization code should work in clustered environment."
 	@priority = "5"
-	test AssertAuthorizationCodeWorksInCluster {
+	test AuthCodeShouldWorkWithClusteredPortals {
```

Comment:
I think we should avoid using short names, as it will confuse others later.
AuthCodeShouldWorkWithClusteredPortals -» AuthorizationCodeShouldWorkWithClusteredPortals